app/src/main/java/com/google/droidjump/MainActivity.java

```diff
@@ -25,13 +30,17 @@
 public class MainActivity extends AppCompatActivity {
 
     private int levelsCount;
-    private int currentLevel;
+    private int currentLevel; // the recent level the user played
+    private int lastLevel; // The biggest level the user played
+    private SharedPreferences gameData;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        gameData = getSharedPreferences(GAME_VIEW_DATA, Context.MODE_PRIVATE);
         levelsCount = GameConstants.GAME_LEVELS_COUNT;
-        currentLevel = 1;
+        currentLevel = getCurrentLevelFromSharedPreferences();
```

Comment:
Do you need currentLevel and lastLevel as fields here at all? Why not to store these values in shared preferences and let them be the source of truth for the whole game?

Reply:
Moved these values to shared preferences.